src/p_range.jl

```diff
@@ -809,39 +800,32 @@ function renumber_partition(partition_in;renumber_local_indices=true)
 end
 
 function local_range(p,np,n,ghost=false,periodic=false)
```

Comment:
Is the new version of this function equivalent to the previous one? In the previous, we distribute the reminder by giving an extra item to the first parts (or the last parts, now not I don't remember). Can you please include the output of this new version for some illustrative input values?

Reply:
The logic on how to distribute the remainder hasn't changed as `offset` is still calculated exactly the same way. I just simplified the function a bit and made it work with more than one layer of ghost atoms.

```julia
julia> [PartitionedArrays.local_range(i, 5, 17) for i in 1:5]'
1×5 adjoint(::Vector{UnitRange{Int64}}) with eltype LinearAlgebra.Adjoint{Int64, UnitRange{Int64}}:
 [1 2 3]  [4 5 6]  [7 8 9]  [10 11 12 13]  [14 15 16 17]

julia> [PartitionedArrays.local_range(i, 5, 17, true) for i in 1:5]'
1×5 adjoint(::Vector{UnitRange{Int64}}) with eltype LinearAlgebra.Adjoint{Int64, UnitRange{Int64}}:
 [1 2 3 4]  [3 4 … 6 7]  [6 7 … 9 10]  [9 10 … 13 14]  [13 14 … 16 17]

julia> [PartitionedArrays.local_range(i, 5, 17, true, true) for i in 1:5]'
1×5 adjoint(::Vector{UnitRange{Int64}}) with eltype LinearAlgebra.Adjoint{Int64, UnitRange{Int64}}:
 [0 1 … 3 4]  [3 4 … 6 7]  [6 7 … 9 10]  [9 10 … 13 14]  [13 14 … 17 18]
```